homeassistant/components/pooldose/sensor.py

```diff
@@ -0,0 +1,219 @@
+"""Sensors for the Seko PoolDose integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import PooldoseConfigEntry, PooldoseCoordinator
+from .entity import PooldoseEntity, device_info
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class PooldoseSensorEntityDescription(SensorEntityDescription):
+    """Describe a PoolDose sensor entity."""
+
+    entity_category: EntityCategory | None = None
+
+
+SENSOR_DESCRIPTIONS: tuple[PooldoseSensorEntityDescription, ...] = (
+    PooldoseSensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        # Unit comes from API (°C or °F)
+    ),
+    PooldoseSensorEntityDescription(
+        key="ph",
+        device_class=SensorDeviceClass.PH,
+        # has no unit
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp",
+        translation_key="orp",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement="mV",
+    ),
+    PooldoseSensorEntityDescription(
+        key="ph_type_dosing",
+        translation_key="ph_type_dosing",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        # has no unit
+    ),
+    PooldoseSensorEntityDescription(
+        key="peristaltic_ph_dosing",
+        translation_key="peristaltic_ph_dosing",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        # has no unit
+    ),
```

Comment:
They have no unit, but they do have possible states, so let's set the device class to ENUM and add the possible states to `options`, this way users know the possible states upfront which makes automating way easier